javascript-algorithms-and-data-structures/learn-intermediate-algorithmic-thinking-by-building-a-dice-game/index.js

```diff
@@ -1,272 +1,259 @@
-    const availableRadioScores = scoreSpans;
-    // I believe this is the first time Object.keys has been introduced
-    const validScoreOptionsKeys = Object.keys(this.validScoreOptions);
-
-    for (let i = 0; i < validScoreOptionsKeys.length; i++) {
-      for (let j = 0; j < availableScoreInputs.length; j++) {
-        if (validScoreOptionsKeys[i] === availableScoreInputs[j].value) {
-          availableScoreInputs[j].disabled = false;
-          availableRadioScores[j].textContent = `, score= ${
-            this.validScoreOptions[availableScoreInputs[j].value]
-          }`;
-        }
-      }
+  // Find the highest count of duplicates that is at least 3 or 4
+  let highestCount = 0;
+  for (const num of arr) {
+    const count = counts[num];
+    if (count >= 3 && count > highestCount) {
+      highestCount = count;
+    }
+    if (count >= 4 && count > highestCount) {
+      highestCount = count;
     }
   }
 
-  isKeepScoreSuccess() {
-    const allScoreInputs = scoreInputs;
+  // Filter the original array to include only the numbers that appear the highest number of times
+  const highestDuplicates = arr.filter((num) => counts[num] === highestCount);
 
-    for (let i = 0; i < allScoreInputs.length; i++) {
-      if (allScoreInputs[i].checked) {
-        const currentValue = allScoreInputs[i].value;
-        const currentScore = this.validScoreOptions[currentValue];
-        this.totalScore += currentScore;
-        this.scoreHistory.push({ [allScoreInputs[i].value]: currentScore });
-        this.currentRound++;
-        this.rollsInCurrentRound = 0;
+  // according to rules, it should be sum of all 5 dices
+  const sumOfAllDices = displayArray.reduce((a, b) => a + b, 0);
 
-        return true;
-      }
-    }
-    alert("Please select a score.");
-    return false;
+  // Return the highest duplicates only if there are at least three or four of them
+  if (highestCount >= 4) {
+    updateRadioOption(1, sumOfAllDices);
+  } else if (highestCount >= 3) {
+    updateRadioOption(0, sumOfAllDices);
+  } else {
+    updateRadioOption(5, 0);
   }
+};
 
-  gameOver() {
-    if (
-      confirm(
-        `You scored ${game.totalScore} points. Do you want to play again?`
-      )
-    )
-      location.reload();
-  }
-}
+/**
+ * ! This will help us detect full house
+ */
+
+const detectFullHouse = (arr) => {
+  // condiiton, 3 of a kind and 2 of a kind
+  const counts = {};
 
-const game = new Game();
+  // Count the occurrences of each number in the array
+  for (const num of arr) {
+    counts[num] = counts[num] ? counts[num] + 1 : 1;
+  }
 
-rollDiceBtn.addEventListener("click", (e) => {
-  if (game.currentRound <= 6) {
-    game.rollDice();
-    game.resetRadioInputs();
-    game.generateValidScoreOptions();
-    game.enableValidScoreInputs();
-    game.updateStatsUI();
-    keepScoreBtn.disabled = false;
+  // Check if there are three of one number and two of another number
+  let hasThreeOfAKind = false;
+  let hasPair = false;
```

Comment:
```suggestion
  // Check if there are three of one number and two of another number
  // Have campers iterate through object first, then refactor this to 
  // use Object.values().includes() instead.
  let hasThreeOfAKind = false;
  let hasPair = false;
```

A note for a cool refactor step.